components/ui/input.tsx

```diff
@@ -1,25 +1,48 @@
 import * as React from "react";
-
 import { cn } from "@/lib/utils";
+import { Eye, EyeOff } from "lucide-react";
 
 export interface InputProps
-  extends React.InputHTMLAttributes<HTMLInputElement> {}
+  extends React.InputHTMLAttributes<HTMLInputElement> {
+  isPassword?: boolean;
+}
 
 const Input = React.forwardRef<HTMLInputElement, InputProps>(
-  ({ className, type, ...props }, ref) => {
+  ({ className, type, isPassword = false, ...props }, ref) => {
+    const [showPassword, setShowPassword] = React.useState(false);
```

Comment:
import useState from react above. That way you won't need to write React.useState each time. 

Also, use useState<boolean>(false);